src/cuckoocache.h

```diff
@@ -428,8 +428,10 @@ class cache
             * for the next iteration.
             */
             last_loc = locs[(1 + (std::find(locs.begin(), locs.end(), last_loc) - locs.begin())) & 7];
-            std::swap(table[last_loc], e);
             // Can't std::swap a std::vector<bool>::reference and a bool&.
+            Element el = std::move(table[last_loc]);
+            table[last_loc] = std::move(e);
+            e = std::move(el);
```

Comment:
Actually I'm a bit baffled by this one - the reason for this change was the instantiation of a bool cache in src/test/fuzz/cuckoocache.cpp, which looks like it would have been built whenever enable-fuzz was true. I don't understand how that ever compiled, but if you don't know the answer offhand I'll investigate.

Reply:
All compilers are broken (as in: They compile wrong code without error, or conversely compile correct code into an incorrect binary). But we should document this brokenness as best as we can, and fix incorrect code separate from feature changes or refactoring or build system changes. Mixing everything into one commit/patchset will make it harder to review and bisect later on.